Angular InstantSearch/routing-seo-friendly/src/app/search/search.component.ts

```diff
@@ -0,0 +1,125 @@
+import { Component } from "@angular/core";
+import { history as historyRouter } from "instantsearch.js/es/lib/routers";
+import algoliasearch from "algoliasearch/lite";
+
+// Returns a slug from the category name.
+// Spaces are replaced by "+" to make
+// the URL easier to read and other
+// characters are encoded.
+function getCategorySlug(name) {
+  return name
+    .split(" ")
+    .map(encodeURIComponent)
+    .join("+");
+}
+
+// Returns a name from the category slug.
+// The "+" are replaced by spaces and other
+// characters are decoded.
+function getCategoryName(slug) {
+  return slug
+    .split("+")
+    .map(decodeURIComponent)
+    .join(" ");
+}
+
+const searchClient = algoliasearch(
+  "B1G2GM9NG0",
+  "aadef574be1f9252bb48d4ea09b5cfe5"
+);
+
+@Component({
+  selector: "app-search",
+  templateUrl: "./search.component.html"
+})
+export class SearchComponent {
+  config = {
+    indexName: "demo_ecommerce",
+    searchClient,
+    routing: {
+      router: historyRouter({
+        windowTitle({ category, query }) {
+          const queryTitle = query ? `Results for "${query}"` : "Search";
+
+          if (category) {
+            return `${category} – ${queryTitle}`;
+          }
+
+          return queryTitle;
+        },
+
+        createURL({ qsModule, routeState, location }) {
+          const urlParts = location.href.match(/^(.*?)\/search/);
```

Comment:
We can't leverage the router of Angular for those operations?

Reply:
done here: 0d935f8